src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -2197,3 +2227,14 @@ type GetResponseFound<TDocument = unknown> = estypes.GetResponse<TDocument> &
 const isFoundGetResponse = <TDocument = unknown>(
   doc: estypes.GetResponse<TDocument>
 ): doc is GetResponseFound<TDocument> => doc.found;
+
+/**
+ * Response headers check to determine if the response is from Elasticsearch
+ * @param headers Response headers
+ * @returns boolean
+ */
+const isEsHeaderValid = (headers: Record<string, unknown> | null) => {
+  return headers && headers['x-elastic-product']
```

Comment:
I'm fairly certain this is the case, but have we validated that the casing of the header names are always normalized to lowercase?

Reply:
Good question! I did do a cursory check but we could handle the case by casting it to lowercase before checking the condition to be safe.